helix-core/src/graphemes.rs

```diff
@@ -28,6 +28,11 @@ pub enum Grapheme<'a> {
 }
 
 impl<'a> Grapheme<'a> {
+    pub fn new_decoration(g: &'static str) -> Grapheme<'a> {
+        assert_ne!(g, "\t");
```

Comment:
Because a tab needs to be turned into `n` spaces (and are based on position).

This could be a debug assert but it doesn't really matter, this is basically meant for constructing constants (it only accepts string constants) so I expect this to be inline and turn either into "always panic" or "never panic"

Reply:
the problem may become clearer if you look at the next line. I pass in `0` for `tab_width` and `visual_x` which are used to calculate real tab size